.github/workflows/lint.yml

```diff
@@ -16,20 +16,28 @@ jobs:
               with:
                 node-version-file: '.nvmrc'
 
-            - name: Cache node modules
-              uses: actions/cache@v3
-              env:
-                cache-name: cache-node-modules
+            - uses: pnpm/action-setup@v2
+              name: Install PNPM
+              id: pnpm-install
               with:
-                path: "**/node_modules"
-                key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+                  version: 8
+                  run_install: false
+            
+            - name: Get pnpm store directory
+              shell: bash
+              run: |
+                echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+
+            - uses: actions/cache@v3
+              name: Setup pnpm cache
+              with:
+                path: ${{ env.STORE_PATH }}
+                key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
                 restore-keys: |
-                  ${{ runner.os }}-build-${{ env.cache-name }}-
-                  ${{ runner.os }}-build-
-                  ${{ runner.os }}-
+                  ${{ runner.os }}-pnpm-store-
```

Comment:
We have a reusable Github action to reduce repetitive code in such scenarios:

https://github.com/prezly/setup-github-actions/